h/feeds/util.py

```diff
@@ -0,0 +1,93 @@
+"""Utility functions for feed-generating code."""
+import cgi
+import urlparse
+
+from dateutil import parser
+
+
+def created_day_string_from_annotation(annotation):
+    """Return a simple created day string for the given annotation.
+
+    Returns a day string like '2015-03-11' from the annotation's 'created'
+    date.
+
+    """
+    return parser.parse(annotation["created"]).strftime("%Y-%m-%d")
+
+
+def tag_uri_for_annotation(annotation, annotation_url):
+    """Return a tag URI (unique identifier) for the given annotation.
+
+    Suitable for use as the value of the <id> element of an <entry> in an
+    Atom feed, or the <guid> element of an <item> in an RSS feed.
+
+    :returns: A tag URI (RFC 4151) for use as the ID for the Atom entry.
+    :rtype: string
+
+    """
+    return u"tag:{domain},{day}:{id_}".format(
+        domain=urlparse.urlparse(annotation_url(annotation)).netloc,
+        day=created_day_string_from_annotation(annotation),
+        id_=annotation["id"])
+
+
+def target_links_from_annotation(annotation):
+    """Return a list of URLs to the given annotation's targets."""
+    links = []
+    targets = annotation.get("target")
+    if isinstance(targets, list):
+        for target in targets:
+            if not isinstance(target, dict):
+                continue
+            source = target.get("source")
+            if source is None:
+                continue
+            links.append(source)
+    return links
+
+
+def description_from_annotation(annotation):
+    """Return a description of the given annotation.
+
+    Suitable for use in the <description> field of an <item> of an RSS feed.
+
+    """
+    def get_selection(annotation):
```

Comment:
Seems like several parts of our code might benefit from pulling this out and just having a normalization/framing function somewhere that ensures everything is a list that should be a list.